plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java

```diff
@@ -335,15 +390,15 @@ protected void buildImage(Dockerfile dockerfile,
         }
     }
 
-    private void pullImage(DockerInstanceKey dockerInstanceKey, final LineConsumer creationLogsOutput) throws MachineException {
-        if (dockerInstanceKey.getRepository() == null) {
+    private void pullImage(DockerMachineSource dockerMachineSource, final LineConsumer creationLogsOutput) throws MachineException {
+        if (dockerMachineSource.getRepository() == null) {
             throw new MachineException("Machine creation failed. Snapshot state is invalid. Please, contact support.");
         }
         try {
             final ProgressLineFormatterImpl progressLineFormatter = new ProgressLineFormatterImpl();
-            docker.pull(dockerInstanceKey.getRepository(),
```

Comment:
I have the same review standards  for all, but I'm not computer to find all the things in all PRs I'm reviewing.
BTW it is you who asked to create new API of docker client that will help to avoid backward incompatible changes. So I don't understand why you are against usage of it.